src/components/nc-slideshow.vue

```diff
@@ -0,0 +1,249 @@
+      :data-slide-to="slideIndex + 1"
+      :style="rightLinkStyle"
+      role="button"
+    >
+      {{ rightLinkText }}
+    </a>
+  </div>
+  
+</template>
+
+<script>
+/**
+ * Code based on https://github.com/myliang/fish-ui/blob/master/src/components/Carousel.vue
+ */
+export default {
+  name: 'nc-slideshow',
+  props: {
+    paginationActiveClass: {
+      type: String,
+      default: 'active'
+    },
+    paginationStyle: {
+      type: Object
+    },
+    hasLinkLeft: {
+      type: Boolean,
+      default: false
+    },
+    hasLinkRight: {
+      type: Boolean,
+      default: false
+    },
+    leftLinkText: {
+      type: String,
+      default: 'Previous'
+    },
+    rightLinkText: {
+      type: String,
+      default: 'Next'
+    },
+    linksDefaultAction: {
+      type: Boolean,
+      default: false
+    },
+    leftLinkStyle: {
+      type: Object
+    },
+    rightLinkStyle: {
+      type: Object
+    },
+    ariaText: {
+      type: Object,
+      default: function() {
+        return {
+          ariaTextDots: 'Slide'
+        }
+      }
+    }
+  },
+  data() {
+    return {
+      slideIndex: 0,
+      slideLength: 0,
+      width: 0,
+      ariaObject: {
+        itemIndex: 'slide',
+        tabPanel: 'tabpanel-0-'
+      },
+      slideList: ''
+    }
+  },
+  methods: {
+    selectNewSlide(index) {
+      this.slideIndex = index
+    },
+    leftLinkHandler(e) {
+      if (this.linksDefaultAction) {
+        this.goToSlide('right')
+      }
+      if (this.slideIndex === 0) {
+        this.$emit('slideshow-first-slide', e)
+      } else {
+        this.$emit('slideshow-click-left-link', e)
+      }
+    },
+    rightLinkHandler(e) {
+      if (this.linksDefaultAction) {
+        this.goToSlide('left')
+      }
+      if (this.slideIndex === this.slideLength - 1) {
+        this.$emit('slideshow-last-slide', e)
+      } else {
+        this.$emit('slideshow-click-right-link', e)
+      }
+    },
+    goToSlide(direction) {
+      if (direction === 'left') {
+        this.slideIndex < this.slideLength - 1
+          ? ++this.slideIndex
+          : this.slideIndex
+      } else {
+        this.slideIndex === 0 ? this.slideIndex : --this.slideIndex
+      }
+    },
+    resizeSlide() {
+      let _this = this
+      this.width = this.$el.offsetWidth
+      this.slideList.forEach(element => {
+        element.style.width = _this.width + 'px'
+      })
+    }
+  },
+  mounted() {
+    window.addEventListener('resize', this.resizeSlide)
+    this.slideList = document.querySelectorAll('.list li')
+    this.slideLength = this.slideList.length || 0
+    this.resizeSlide()
+  },
+  beforeDestroy() {
+    window.removeEventListener('resize', this.resizeSlidel)
+  }
+}
+</script>
+
+<style lang="scss" scoped>
+$dotBackgrounColor: #d8d8d8;
+$dotBackgrounColorActive: #ff7f3f;
+.nc-slideshow {
+  position: relative;
+  display: block;
+  box-sizing: border-box;
+  height: 100%;
+  &__content {
```

Comment:
Sería conveniente dejar espacios entre reglas para ayudar a la lectura
```suggestion

  &__content {
```

Reply:
Modificado